File: proj_firestore.js
const admin = require('firebase-admin');

var serviceAccount = require('/home/pi/hnr/my-awesome-project-9a50a-firebase-adminsdk-l7ym1-53abc2353b.json');

admin.initializeApp({
  credential: admin.credential.cert(serviceAccount)
});


var db = admin.firestore();

var docRef = db.collection('users').doc('alovelace');

var setAda = docRef.set({
  first: 'Ada',
  last: 'Lovelace',
  born: 1815
});

var aTuringRef = db.collection('users').doc('aturing');

var setAlan = aTuringRef.set({
  'first': 'Alan',
  'middle': 'Mathison',
  'last': 'Turing',
  'born': 1912
});

aTuringRef.update({
	age: 'dead'
})

db.collection('users').get()
  .then((snapshot) => {
    snapshot.forEach((doc) => {
      console.log(doc.id, '=>', doc.data());
    });
  })
  .catch((err) => {
    console.log('Error getting documents', err);
  });

File: proj_firestore.py
import firebase_admin
from firebase_admin import credentials
from firebase_admin import firestore


cred = credentials.Certificate('/home/pi/hnr/my-awesome-project-9a50a-firebase-adminsdk-l7ym1-53abc2353b.json')
firebase_admin.initialize_app(cred)


db = firestore.client()

doc_ref = db.collection(u'users').document(u'alovelace')
doc_ref.set({
    u'first': u'Ada',
    u'last': u'Lovelace',
    u'born': 1815
})

doc_ref = db.collection(u'users').document(u'aturing')
doc_ref.set({
    u'first': u'Alan',
    u'middle': u'Mathison',
    u'last': u'Turing',
    u'born': 1912
})
